elks/arch/i86/kernel/timer.c

```diff
@@ -79,6 +79,15 @@ void timer_tick(int irq, struct pt_regs *regs)
     }
 #endif
 
+#ifdef CONFIG_ARCH_SWAN
+    /* spin timer segments */
+    if (spin_on && !(jiffies & 7)) {
+        static unsigned char wheel[4] = {0x00, 0x20, 0x10, 0x08};
+        static unsigned char c = 0;
+
+        outb((inb(0x15) & 0x07) | wheel[c++ & 0x03], 0x15);
+    }
+#else
```

Comment:
Do you really want this in here? This little bit of fancy is only called from the BIOS block device driver, so I'm wondering whether it will ever run on Swan. It's purpose is to spin a visual while waiting for very slow disk I/O.

If you think that Swan will not use this feature or have any very slow I/O, I would remove it and the similar code below in spin_timer().